subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/LoggingIntegrationTest.groovy

```diff
@@ -153,94 +167,76 @@ class LoggingIntegrationTest extends AbstractIntegrationTest {
-    void infoLogging() {
-        checkOutput(this.&run, logOutput.info)
-    }
-
-    @Test
-    @ToBeFixedForConfigurationCache(skip = ToBeFixedForConfigurationCache.Skip.FAILS_TO_CLEANUP)
-    void debugLogging() {
-        checkOutput(this.&run, logOutput.debug)
-    }
-
-    @Test @UsesSample('tutorial/logging/groovy')
-    void sampleQuietLogging() {
-        checkOutput(this.&runSample, sample.quiet)
-    }
-
-    @Test @UsesSample('tutorial/logging/groovy')
-    void sampleLifecycleLogging() {
-        checkOutput(this.&runSample, sample.lifecycle)
-    }
-
-    @Test @UsesSample('tutorial/logging/groovy')
-    void sampleInfoLogging() {
-        checkOutput(this.&runSample, sample.info)
-    }
-
-    @Test @UsesSample('tutorial/logging/groovy')
-    void sampleDebugLogging() {
-        checkOutput(this.&runSample, sample.debug)
-    }
+    @ToBeFixedForConfigurationCache(because = "https://github.com/gradle/gradle/issues/25483")
+    def "build emits #level logging"() {
+        LogLevel logLevel = logOutput."$level"
+        resources.maybeCopy('LoggingIntegrationTest/logging')
+        TestFile loggingDir = testDirectory
+        loggingDir.file("buildSrc/build/.gradle").deleteDir()
+        loggingDir.file("nestedBuild/buildSrc/.gradle").deleteDir()
 
-    @Test
-    void multiThreadedQuietLogging() {
-        checkOutput(this.&runMultiThreaded, multiThreaded.quiet)
-    }
+        String initScript = new File(loggingDir, 'init.gradle').absolutePath
+        List<String> allArgs = logLevel.args + ['-I', initScript]
 
-    @Test
-    void multiThreadedlifecycleLogging() {
-        checkOutput(this.&runMultiThreaded, multiThreaded.lifecycle)
-    }
+        when:
+        executer.noExtraLogging().inDirectory(loggingDir).withArguments(allArgs)
+        run "log"
+        then:
+        logLevel.checkOuts(result)
 
-    @Test
-    void multiThreadedDebugLogging() {
-        checkOutput(this.&runMultiThreaded, multiThreaded.debug)
+        where:
+        level << ['quiet', 'lifecycle', 'info', 'debug']
     }
 
-    def run(LogLevel level) {
+    @ToBeFixedForConfigurationCache(because = "https://github.com/gradle/gradle/issues/24182")
+    def "custom logger emits #level logging"() {
+        LogLevel logLevel = customLoggerOutput."$level"
         resources.maybeCopy('LoggingIntegrationTest/logging')
         TestFile loggingDir = testDirectory
         loggingDir.file("buildSrc/build/.gradle").deleteDir()
         loggingDir.file("nestedBuild/buildSrc/.gradle").deleteDir()
 
         String initScript = new File(loggingDir, 'init.gradle').absolutePath
-        String[] allArgs = level.args + ['-I', initScript]
-        return executer.noExtraLogging().inDirectory(loggingDir).withArguments(allArgs).withTasks('log').run()
+        String customLoggerInitScript = new File(loggingDir, 'custom-logger-init.gradle').absolutePath
+        List<String> allArgs = logLevel.args + ['-I', initScript] + ['-I', customLoggerInitScript]
```

Comment:
Ah, because only in this test you wanted the configuration made by the custom logging init script. Maybe make it explicit?

Reply:
I separated the tests: One does test general logging that includes logging from init scripts, and the other one checks that under the presence of a custom logger. So I split the registration of the custom logger into a separate init script.

Is that clear already from the test names? Where'd I add the clarifying comment?

Reply:
I take it back, I think this is fine as is. My doubt came from looking at this as a change set/diff, instead of trying to make sense of the final code as a whole.